core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala

```diff
@@ -462,7 +462,10 @@ private[ui] class JobDataSource(
       }
       case "Description" => new Ordering[JobTableRowData] {
         override def compare(x: JobTableRowData, y: JobTableRowData): Int =
-          Ordering.String.compare(x.lastStageDescription, y.lastStageDescription)
+          (x.lastStageDescription, y.lastStageDescription) match {
```

Comment:
I knew this function seemed bulky, I'll go through and update it as you suggested.